agent/connect/ca/provider_consul.go

```diff
@@ -487,8 +471,12 @@ func (c *ConsulProvider) SignIntermediate(csr *x509.CertificateRequest) (string,
 	effectiveNow := time.Now().Add(-1 * CertificateTimeDriftBuffer)
 	template := x509.Certificate{
```

Comment:
This blob threw me until I found it.

This spot in the code _accepts_ a CSR but does not pass that CSR off to the `crypto/x509` package or anything of the sort. The incoming CSR has select fields plucked out of it to assign to the newly generated intermediate cert.

I done an overmatch during cleanup and stripped the `Subject: csr.Subject,` line from this function which super broke intermediate certs (which only put data in the Subject/CN field). For now I modified this to handle this more like how a CSR should operate and copy the various SAN fields along with the Subject and any provided ExtraExtensions as-is.

We can discuss if the CA certs themselves should be using SAN fields, too, but the CN fields for those things are less awful to construct so it's fine to keep them for now.

Reply:
FWIW, if we're not going to use CN everywhere else, it probably makes sense to stop using it for the CA and reflect that in the TLS utilities. Is there a reason not to?

Reply:
AWS PCA requires a CN field to be populated in order to create the Root CA.

Reply:
Interesting! Seems to align with that I'm finding in https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.6 and https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4 where the Subject/CN is required.